### PR TITLE
[BarClerk-945] - [FE] Integrate Create, Update and Get Court Appearan…

### PR DESCRIPTION
**Pull request by jasonclchua**
…ces API

## Issue Link
- https://app.asana.com/0/1203234368773394/1203256371568945

## Definition of Done
- [ ] User is able to create a CourtAppearance for a Client
- [ ] User is able to view the CourtAppearances of a Client under Court Appearance Page
- [ ] User is able to edit the CourtAppearance of a Client

## Notes
- [ ] the columns (court, judicial_officer, orders, and other_notes) under court_appearance table has been updated to be nullable.

## Pre-condition
- [ ] run `php artisan migrate` to update the court_appearances table columns to have nullable() values
- [ ] add a Matter under the main dashboard and visit the newly added record

* Create a new Court Appearance
- [ ] go to the CourtAppearances tab and create a new court appearance record for this Client
  - [ ] the date, next_court_date, and time are required fields.
- [ ] upon creation of new CourtAppearance, the page should render the new record
* Update a new Court Appearance
- [ ] edit the court appearance record by clicking on the pencil_icon on the right side of the accordion
- [ ] fill in the updated input fields and save
  - [ ] the date, next_court_date, and time are required fields.
- [ ] upon updating the CourtAppearance card, the page should render with the updated record.

## Screenshots/Recordings
[CourtAppearance_Barclerk.webm](https://user-images.githubusercontent.com/118783823/206848064-3ed88576-cd2e-4c10-843e-bfd735d5c674.webm)

### CODE REVIEW

**Review by acatzk — APPROVED** [review]
# Approved 💯

**Review by abduljalilpalala — CHANGES_REQUESTED** [review]
Add a message when there is no data in the list.
Sample: "No Data Available"

Add a message when an error has occured
Sample: "Ooops! Something went wrong!"

Add a label which indicates the total number of court appearances

File: client/src/components/molecules/CourtAppearanceList/index.tsx [hunk 1]
```diff
@@ -1,35 +1,46 @@
...
 
-  const handlePageChange = ({ selected }: { selected: number }): void => setPageNumber(selected)
+  const handlePageChange = async ({ selected }: { selected: number }) => {
+    await dispatch(fetchClientCourtAppearances({ clientId: Number(query?.id), page: selected + 1 }))
+  }
 
   return (
     <>
-      {displayCourtAppearances?.map((courtAppearance) => {
+      {courtAppearancesList?.map((courtAppearance: any) => {
```
Comment (jsvelte) [inline client/src/components/molecules/CourtAppearanceList/index.tsx]: If you have a proper type for `courtAppearancesList` you don't need to add the :any during iteration but then I will prove this one

File: client/src/shared/validation/index.ts [hunk 2]
```diff
@@ -60,13 +60,13 @@ export const MatterFormSchema = Yup.object().shape({
...
 
 export const CourtAppearanceSchema = Yup.object().shape({
-  date: Yup.string().max(255).required('Must be a valid date'),
-  next_court_date: Yup.string().max(255).required('Must be a valid date'),
-  time: Yup.string().max(255).required('Must be a valid time'),
-  court: Yup.string().max(255).required('This field is required'),
-  judicial_officer: Yup.string().max(255).required('This field is required'),
-  orders: Yup.string().required('This field is required'),
-  other_notes: Yup.string()
+    date: Yup.string().required('This field is required'),
```
Comment (jsvelte) [inline client/src/shared/validation/index.ts:63]: Date is required

File: client/src/shared/validation/index.ts [hunk 3]
```diff
@@ -60,13 +60,13 @@ export const MatterFormSchema = Yup.object().shape({
...
+    next_court_date: Yup.string().required('This field is required'),
```
Comment (jsvelte) [inline client/src/shared/validation/index.ts:64]: Next Court Date is required

File: client/src/shared/validation/index.ts [hunk 4]
```diff
@@ -60,13 +60,13 @@ export const MatterFormSchema = Yup.object().shape({
...
+    time: Yup.string().required('This field is required'),
```
Comment (jsvelte) [inline client/src/shared/validation/index.ts:65]: Time is required

**Review by jsvelte — APPROVED** [review]
I left some comment and then # Approved 💯

File: client/src/components/molecules/CourtAppearanceList/index.tsx (hunk 1 above)
Comment (abduljalilpalala) [reply client/src/components/molecules/CourtAppearanceList/index.tsx]: @jasonclchua Just remove `any` from this one.

File: client/src/shared/validation/index.ts (hunk 4 above)
Comment (abduljalilpalala) [reply client/src/shared/validation/index.ts:65]: @jsvelte It's already indicated the the `time` field is required

File: client/src/shared/validation/index.ts (hunk 3 above)
Comment (abduljalilpalala) [reply client/src/shared/validation/index.ts:64]: @jsvelte It's already indicated the the `next_court_date` field is required

File: client/src/shared/validation/index.ts (hunk 2 above)
Comment (abduljalilpalala) [reply client/src/shared/validation/index.ts:63]: @jsvelte It's already indicated the the `date` field is required

**Review by abduljalilpalala — APPROVED** [review]
### Approved! ✔️

**Pull request merged**